mlflow/projects/__init__.py

```diff
@@ -734,13 +790,13 @@ def _build_docker_image(work_dir, project, active_run):
     return tag_name
 
 
-def _get_docker_tag_name(project_name, work_dir):
+def _get_docker_tag_name(imagename, work_dir):
     """Returns an appropriate Docker tag for a project based on name and git hash."""
-    project_name = project_name if project_name else "docker-project"
+    imagename = imagename if imagename else "docker-project"
     # Optionally include first 7 digits of git SHA in tag name, if available.
     git_commit = _get_git_commit(work_dir)
-    version_string = "-" + git_commit[:7] if git_commit else ""
-    return "mlflow-" + project_name + version_string
+    version_string = ":" + git_commit[:7] if git_commit else ""
+    return "mlflow-" + imagename + version_string
```

Comment:
If I set my `image-uri` to something like `"registry.dev.databricks.com/andrew/mlflow-test"` this won't work since it changes the host to `mlflow-registry.dev.databricks.com`. How are you using `image-uri` to get it to work?

Reply:
Sorry, my mistake, I should have ran it here. This issue was introduced in another PR, the [PR 1331](https://github.com/mlflow/mlflow/pull/1331), I've just pulled it so this branch could be able to merge to master.

Done.